store/modules/stimulation/getters.js

```diff
@@ -41,18 +49,29 @@ export default {
   },
 };
 
-const get_protocol_editor_color = (list) => {
-  const color = generate_random_color(false);
-  const duplicate_color = list.filter((protocol) => protocol.color === color).length > 0;
-
-  return duplicate_color ? get_protocol_editor_color(list) : color;
-};
-
 // TODO Luci, handle if there are more than 26 protocols
 const get_protocol_editor_letter = (list) => {
-  const current_protocol_assignment = list[list.length - 1].letter;
-  const current_alphabet_idx = alphabet.indexOf(current_protocol_assignment);
-  return alphabet[current_alphabet_idx + 1];
-};
+  // grab just the first letter in case of AA BB CCC DDDD, unlikely but just in case.
+  const current_protocol_assignment = list[list.length - 1].letter[0];
+
+  const current_alphabet_idx = ALPHABET.indexOf(current_protocol_assignment);
+  let letter_assignment = ALPHABET[current_alphabet_idx + 1];
```

Comment:
I think you could incorporate the module operator here as well to simplify things